benches/lib.rs

```diff
@@ -70,3 +77,71 @@ fn bench_lib_merge_results(b: &mut Bencher) {
 
     b.iter(|| black_box(grcov::merge_results(&mut result, result2.clone())));
 }
+
+#[bench]
+fn bench_lib_consumer(b: &mut Bencher) {
+    let num_threads = 2;
+    let result_map: Arc<SyncCovResultMap> = Arc::new(Mutex::new(FxHashMap::with_capacity_and_hasher(20_000, Default::default())));
+    let (sender, receiver) = unbounded();
+    let source_root = None;
+    let working_dir = PathBuf::from("");
+    let gcno_buf: Vec<u8> = vec![
+        111, 110, 99, 103, 42, 50, 48, 52, 74, 200, 254, 66, 0, 0, 0, 1, 9, 0, 0, 0, 0, 0, 0,
+        0, 236, 217, 93, 255, 2, 0, 0, 0, 109, 97, 105, 110, 0, 0, 0, 0, 2, 0, 0, 0, 102, 105,
+        108, 101, 46, 99, 0, 0, 1, 0, 0, 0, 0, 0, 65, 1, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 67, 1, 3, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 67, 1, 3,
+        0, 0, 0, 1, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 69, 1, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 69, 1, 8, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 102,
+        105, 108, 101, 46, 99, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0,
+    ];
+
+    b.iter(|| {
```

Comment:
This will benchmark thread creation too, but I hope it's an insignificant part of the runtime (given that you're sending a lot of messages).

Reply:
yeah I tried to remove this part from b(...) but to measure something we need to join in b and so we need to have parsers in b but parsers is not clonable because its content are not.